core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -408,7 +408,8 @@ private[spark] class ExecutorAllocationManager(
       executorsPendingToRemove.add(executorId)
       true
     } else {
-      logWarning(s"Unable to reach the cluster manager to kill executor $executorId!")
+      logWarning(s"Unable to reach the cluster manager to kill executor $executorId!" +
+        s"Or no executor to kill!")
```

Comment:
1. or maybe we need to change that to "or executor is not eligible to kill". since there are two situations to get empty list: a) busy b) kill that repeatedly

2. There is not action to take in other parts of the code. The problem here is if we not change the API (return type). it is really hard to tell it is not acknowledged or no executor to kill. If it is not acknowledged, we should print warning, but if no executor to kill, it is the normal case with info level.